php/admin/migration/class-post-type.php

```diff
@@ -84,20 +85,23 @@ public function __construct( $previous_post_type_slug, $previous_block_namespace
 	 * These each store a config for a custom block,
 	 * they aren't blocks that users entered into the block editor.
 	 *
-	 * @return array The migration result: counts of success and errors.
+	 * @return array|WP_Error An array on success, a WP_Error on failure.
 	 */
 	public function migrate_all() {
-		$posts         = $this->query_for_posts();
-		$success_count = 0;
-		$error_count   = 0;
+		$posts              = $this->query_for_posts();
+		$success_count      = 0;
+		$error_count        = 0;
+		$max_allowed_errors = 20;
+		$errors             = new WP_Error();
 
-		while ( ! empty( $posts ) ) {
+		while ( ! empty( $posts ) && $error_count < $max_allowed_errors ) {
```

Comment:
Like with the post content, this now exits if there are 20 or more errors.

Reply:
Like you brought up before, if there are several errors, this can be an infinite loop.